src/Tuura/Plato/Translation.hs

```diff
@@ -89,21 +89,27 @@ validateInterface signs circuit
 
 -- Perform cartesian product on list of lists. This will also sort and remove
 -- duplicates in sublists, and remove supersets for the most compact form.
-cartesianProduct :: Ord a => NonEmpty.NonEmpty [a] -> [[a]]
+cartesianProduct :: Ord a => NonEmpty.NonEmpty [Transition a] -> [[Transition a]]
 cartesianProduct l = removeSupersets sortAllLists
   where
     sequenced    = sequence (NonEmpty.toList l)
     removeDupes  = map nub sequenced
-    sortAllLists = map sort removeDupes
+    removeRedun  = removeRedundancies removeDupes
+    sortAllLists = map sort removeRedun
 
 -- Sort list of lists from largest length to shortest, then remove any lists
 -- that have shorter subsequences within the rest of the list.
-removeSupersets :: Eq a => [[a]] -> [[a]]
+removeSupersets :: Eq a => [[Transition a]] -> [[Transition a]]
 removeSupersets s = [ x | (x:xs) <- tails sortByLength, not (check x xs) ]
   where
     check current = any (`isSubsequenceOf` current)
     sortByLength  = sortBy (comparing $ negate . length) s
 
+removeRedundancies :: Eq a => [[Transition a]] -> [[Transition a]]
+removeRedundancies s = filter (\ts -> 
```

Comment:
You can drop `\ts -> ` and ` ts` here. And then the same with ` s`. (This is called eta reduction/conversion: https://wiki.haskell.org/Eta_conversion).